### ISSUE TITLE
Add support for healthcheck endpoint

### ISSUE THREAD

**Issue by varet80** [LibrePhotos/librephotos#495]
In order to support all modern infrastructure, natively (kubernetes). A health endpoint is a good solution

We can automate the readiness of a service! and also get instance health check through the service itself.

An example of how this endpoint would work is:
Every request to /healthz end point would return the status of Postgresql and Redis connectivity!
I would suggest 4 options:
- `/healthz`: returns status and 200 code even if any of the postgresql and/or redis are not ready! this could show the Backend is reachable!
- `/healthz/postgresql`: Will return a non 200 code, if postgresql is not accessible
- `/healthz/redis`: Will return a non 200 code, if redis is not accessible
- `/healthz/ready`: Will return a non 200 code, if either redis or postgresql is not accessible

With the above options I could use:
- `/healthz` - for startup/liveness probe in kubernetes
- `/healthz/ready` - for readinessprobe

**Comment by sickelap** [comment]
I am pretty sure that redis and postgres will be ready by the time application (in current state) will be able to respond to health or readiness request. So don't think it's necessary to add those to health check. I think simple /health (or /healthz) should be enough to know when the backend is up anready to go.

**Comment by varet80** [comment]
Makes sense. Although the difference betweeen healthz and having redis/postgresql is that.
If you have a healthcheck showing that Redis and/or pG works, you can make the container unready (without restarting the pod) while there is an outage in PG/Redis! this way it follows the best practices on k8s.
